SEP-0012.md

```diff
@@ -0,0 +1,553 @@
+`NDCube` also provides simple summary visualisation support included in the object.
+This uses matplotlib to produce quick, coordinate-aware, and interactive visualisations of the data.
+The final major component added in `NDCube` is the concept of "extra coords" which allows the specification and integration with coordinates describing the data not specified by the primary WCS for the cube.
+
+## ndcube 2
+
+ndcube 1.0 was built on the functionality which existed in astropy at the time and provides a very useful foundation for tools such as the sunraster package as well as other efforts to handle multi-dimensional data in Python.
+During the process of writing NDCube 1.0 a lot of potential functionality which could be added to astropy to improve the WCS support was identified.
+In addition to this in-between the release of ndcube 1.0 and the development of 2.0, various long running projects such as the acceptance and implementation of [APE 14](https://github.com/astropy/astropy-APEs/blob/master/APE14.rst) were completed.
+This means that once many of these brand new features had been implemented in Astropy and gWCS it provided an excellent opportunity to reassess the state of NDCube and to rebase it onto this new WCS functionality.
+
+The main motivation for updating ndcube to use the APE 14 interface was to provide support for gWCS which is being used by the DKIST user tools.
+A lot of the other changes presented in this SEP are a consequence of this decision.
+
+
+## Summary of Changes from ndcube 1
+
+The fundamental change since the original version of ndcube is that the WCS type has been changed to be the High Level API as described by astropy's APE 14.
+This opens a lot of possibilities to the future uses of `NDCube` as the APE 14 API allows the mapping between pixel and world coordinates to be described by almost any functional transform.
+This could be the commonly used FITS WCS standard, or gWCS or custom implementations for specific facilities with different requirements.
+
+Having made this change, which was a large breaking change from the original API, we reviewed all the existing functionality and API as it was adapted to use the high level WCS interface and the result is presented below.
+
+# Detailed Description
+
+The API presented here is limited to the three core classes `NDCubeABC`, `ExtraCoordsABC` and `GlobalCoordsABC`.
+This is primarily an attempt to limit the discussion in this SEP to the core API and not implementation decisions taken in the development of ndcube.
+If people are interested in the implementation of the proposed API large amounts of it are implemented on the master branch of the [ndcube repository](https://github.com/sunpy/ndcube) or open pull requests.
+
+The `NDCube` class inherits the astropy [`NDDataBase`](https://github.com/astropy/astropy/blob/master/astropy/nddata/nddata_base.py#L11) class, including inheriting the requirement that the `.wcs` property be a `BaseHighLevelWCS` instance.
```

Comment:
Updating the link to "today's" version of `master` in astropy as that ref may change in the future.
```suggestion
The `NDCube` class inherits the astropy [`NDDataBase`](https://github.com/astropy/astropy/blob/e4405974c1d78f48230a6736a25286fb04407527/astropy/nddata/nddata_base.py#L11) class, including inheriting the requirement that the `.wcs` property be a `BaseHighLevelWCS` instance.
```